lumeer-storage/lumeer-storage-mongodb/src/main/java/io/lumeer/storage/mongodb/dao/project/MorphiaCollectionDao.java

```diff
@@ -140,8 +141,18 @@ public Collection getCollectionByCode(final String code) {
       if (searchQuery.isFulltextQuery()) {
          mongoQuery.search(searchQuery.getFulltext());
       }
+      Set<ObjectId> collectionIds = searchQuery.getCollectionIds() != null ? searchQuery.getCollectionIds().stream().map(ObjectId::new).collect(Collectors.toSet()) : Collections.emptySet();
       if (searchQuery.isCollectionCodesQuery()) {
-         mongoQuery.field(MorphiaCollection.CODE).in(searchQuery.getCollectionCodes());
+         if (!collectionIds.isEmpty()) {
+            mongoQuery.or(
```

Comment:
Why this if statement? We use OR in the database query so we can always run it with both `code` and `id` conditions. Cannot we?

Reply:
It doesn't make sense to use or condition with only one statement (if second one is empty)

Reply:
Ok, I have finally understood the code. But this bunch of `if` statements look really complicated and is hard to understand at first sight.